contrib/tvmop/core/fromnumeric.py

```diff
@@ -0,0 +1,203 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# coding: utf-8
+import tvm
+import topi
+from .. import defop, AllTypes, AllTypesButHalf
+
+def kernel_cumprod(X, ishape, dtype, ndim, axis):
```

Comment:
Suggestion: Maybe we can make this a common scan kernel, so that in the future similar scan operators could re-use this.

Reply:
Good. I will do it. What about moving the generalized kernel to `contrib/tvmop/utils.py`?